### PR TITLE
cptx: push gRPC config along with cert

### PR DESCRIPTION
**Pull request by nitinsoniism**
* make sure gRPC service starts after configuring certs
* modify UT
* remove gRPC config from default config
* fix hostname
* add cptx and host container image config

### CODE REVIEW

File: examples/juniper/cptx-host/juniper.config [hunk 1]
```diff
@@ -6,16 +6,6 @@ system {
         ssh {
             root-login allow;
         }
-        extension-service {
```
Comment (marcushines) [inline examples/juniper/cptx-host/juniper.config:9]: why are you removing this config

File: examples/juniper/cptx-host/juniper.config [hunk 2]
```diff
@@ -36,17 +26,3 @@ interfaces {
         }
     }
 }
-openconfig-system:system {
```
Comment (marcushines) [inline examples/juniper/cptx-host/juniper.config:39]: same

File: topo/node/cptx/cptx.go [hunk 3]
```diff
@@ -140,6 +140,26 @@ func (n *Node) GenerateSelfSigned(ctx context.Context) error {
...
+		"set system services extension-service request-response grpc ssl use-pki",
+		"set openconfig-system:system openconfig-system-grpc:grpc-servers grpc-server grpc-server config services GNMI",
+		"set openconfig-system:system openconfig-system-grpc:grpc-servers grpc-server grpc-server config enable true",
+		"set openconfig-system:system openconfig-system-grpc:grpc-servers grpc-server grpc-server config port 32767",
+		"set openconfig-system:system openconfig-system-grpc:grpc-servers grpc-server grpc-server config transport-security true",
+		"set openconfig-system:system openconfig-system-grpc:grpc-servers grpc-server grpc-server config certificate-id grpc-server-cert",
+		"set openconfig-system:system openconfig-system-grpc:grpc-servers grpc-server grpc-server config listen-addresses 0.0.0.0",
+		"commit",
+	}
+	resp, err := n.cliConn.SendConfigs(cfgs)
```
Comment (marcushines) [inline topo/node/cptx/cptx.go:154]: why are you sending this now as cli vs. in the initial config?

File: examples/juniper/cptx-host/cptx-host.pb.txt [hunk 4]
```diff
@@ -5,12 +5,13 @@ nodes: {
     model: "cptx"
     os: "evo"
     config: {
+      image: "cptx:test"
```
Comment (marcushines) [inline examples/juniper/cptx-host/cptx-host.pb.txt:8]: why add this?

File: examples/juniper/cptx-host/cptx-host.pb.txt [hunk 5]
```diff
@@ -5,12 +5,13 @@ nodes: {
...
       config_path: "/home/evo/configdisk"
       config_file: "juniper.conf"
       file: "juniper.config"
       cert: {
             self_signed: {
-                cert_name: "ca-ipsec",
+                cert_name: "grpc-server-cert",
```
Comment (marcushines) [inline examples/juniper/cptx-host/cptx-host.pb.txt:14]: why rename this?

File: examples/juniper/cptx-host/cptx-host.pb.txt (hunk 4 above)
Comment (nitinsoniism) [reply examples/juniper/cptx-host/cptx-host.pb.txt:8]: Just to keep this in sync with the other cptx example cptx-ixia/cptx-ixia.pb.txt

File: examples/juniper/cptx-host/cptx-host.pb.txt (hunk 5 above)
Comment (nitinsoniism) [reply examples/juniper/cptx-host/cptx-host.pb.txt:14]: ca-ipsec was not the apt name for this certificate as this is used for grpc

File: examples/juniper/cptx-host/juniper.config (hunk 1 above)
Comment (nitinsoniism) [reply examples/juniper/cptx-host/juniper.config:9]: Having this config part of the default config causes a race where gRPC service is started before the certer interface is invoked leading to failure. This should be configured after self-signed certs are configured.

File: topo/node/cptx/cptx.go (hunk 3 above)
Comment (nitinsoniism) [reply topo/node/cptx/cptx.go:154]: Having this config part of the default config causes a race where gRPC service is started before the certer interface is invoked leading to failure. This should be configured after self-signed certs are configured.

File: examples/juniper/cptx-host/cptx-host.pb.txt (hunk 4 above)
Comment (alexmasi) [reply examples/juniper/cptx-host/cptx-host.pb.txt:8]: Can you remove both this and the image field in cptx-ixia/cptx-ixia.pb.txt?

I would prefer to keep the examples using the node implementation defaults unless there is a specific version required to run the example (which in this case there is not)

Comment (nitinsoniism) [reply examples/juniper/cptx-host/cptx-host.pb.txt]: makes sense. Done.

**Review by marcushines — DISMISSED** [review]

**Comment by nitinsoniism** [comment]
@marcushines @alexmasi could you please hold onto merging this for a bit? I need to clarify something before we merge this.

**Comment by nitinsoniism** [comment]
I will send out another commit part of the same PR in a bit.

**Comment by alexmasi** [comment]
Sounds good, I will also test this manually to make sure it works on our end as well before merging

**Comment by nitinsoniism** [comment]
@alexmasi I am done with all my commits. You can try this out and let me know if you see any issues.

**Comment by alexmasi** [comment]
Thanks, I'll give this a manual run tomorrow to confirm

**Comment by alexmasi** [comment]
@nitinsoniism All looks good to me, however one question on the README. For gRIBI my output did not line up with the guide, I got:

```
$ gribic -a 192.168.11.50:9340 -u root -p Google123 --skip-verify get --ns default --aft ipv4
INFO[0000] target 192.168.11.50:9340: final get response:
INFO[0000] got 1 results
INFO[0000] "192.168.11.50:9340":
```

Is this a typo in the guide or is my config missing something?

**Comment by nitinsoniism** [comment]
> @nitinsoniism All looks good to me, however one question on the README. For gRIBI my output did not line up with the guide, I got:
> 
> ```
> $ gribic -a 192.168.11.50:9340 -u root -p Google123 --skip-verify get --ns default --aft ipv4
> INFO[0000] target 192.168.11.50:9340: final get response:
> INFO[0000] got 1 results
> INFO[0000] "192.168.11.50:9340":
> ```
> 
> Is this a typo in the guide or is my config missing something?

Hey @alexmasi, thanks for trying this out. Yes, you will see a different result for gRIBI. This was just an example output I pasted in the README. The RIB entry was programmed using one of the ondatra scripts prior to running the gribic command. This wouldn't be part of the default config. One can use gribic too to program a RIB entry and it will display the same result.

**Review by alexmasi — APPROVED** [review]

**Pull request merged**